src/CoreWCF.RabbitMQ.Client/src/CoreWCF/ServiceModel/Channels/RabbitMqChannelFactory.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Linq;
+using System;
+using System.ServiceModel.Channels;
+using System.Threading.Tasks;
+
+namespace CoreWCF.ServiceModel.Channels
+{
+    public class RabbitMqChannelFactory : ChannelFactoryBase<IOutputChannel>
+    {
+        private BufferManager _bufferManager;
+        private MessageEncoderFactory _messageEncoderFactory;
+        private RabbitMqTransportBindingElement _transport;
+
+        internal RabbitMqChannelFactory(RabbitMqTransportBindingElement transport, BindingContext context)
+            : base(context.Binding)
+        {
+            _transport = transport;
+            _bufferManager = BufferManager.CreateBufferManager(transport.MaxBufferPoolSize, int.MaxValue);
+
+            var messageEncoderBindingElements = context.BindingParameters.OfType<MessageEncodingBindingElement>().ToList();
+            if (messageEncoderBindingElements.Count > 1)
+            {
+                throw new InvalidOperationException("More than one MessageEncodingBindingElement was found in the BindingParameters of the BindingContext");
+            }
+
+            if (messageEncoderBindingElements.Count == 1)
+            {
+                _messageEncoderFactory = messageEncoderBindingElements.First().CreateMessageEncoderFactory();
+            }
+            else
+            {
+                _messageEncoderFactory = RabbitMqDefaults.DefaultMessageEncoderFactory;
+            }
+        }
+
+        public RabbitMqTransportBindingElement Transport => _transport;
+
+        public BufferManager BufferManager => _bufferManager;
+
+        public MessageEncoderFactory MessageEncoderFactory => _messageEncoderFactory;
+
+        public override T GetProperty<T>()
+        {
+            T messageEncoderProperty = MessageEncoderFactory.Encoder.GetProperty<T>();
+            if (messageEncoderProperty != null)
+            {
+                return messageEncoderProperty;
+            }
+
+            if (typeof(T) == typeof(MessageVersion))
+            {
+                return (T)(object)MessageEncoderFactory.Encoder.MessageVersion;
+            }
+
+            return base.GetProperty<T>();
+        }
+
+        protected override void OnOpen(TimeSpan timeout)
+        {
+        }
+
+        protected override IAsyncResult OnBeginOpen(TimeSpan timeout, AsyncCallback callback, object state)
+        {
+            return Task.CompletedTask;
+        }
```

Comment:
`Task.CompletedSynchronously` always returns false, which tells the caller that the callback and state will get called by the completion of OnBeginOpen, which has already happened as `Task.IsCompleted` returns true. You would need to do a null check on the callback and call it passing the `IAsyncResult` that you are returning. The problem with that is the `IAsyncResult` you return needs to wrap the `state` object you passed in, and `Task.CompeltedTask` doesn't do that. All this is to say there are no shortcuts when it comes to implementing APM methods (which is why it was thankfully replaced with async/await) so you're best off copying in the `TaskHelpers.ToApm` and `TaskHelpers.ToApmEnd` helper methods and using them to make sure everything is done right.  
I noticed you didn't override OnBeginClose/OnEndClose, so I think ChannelFactoryBase handles them. Do you need to override the Open methods too? I think you might be able to just remove these methods.